rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/tencent/TencentBakeHandler.groovy

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2015 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.providers.tencent
+
+import com.netflix.spinnaker.rosco.api.Bake
+import com.netflix.spinnaker.rosco.api.BakeOptions
+import com.netflix.spinnaker.rosco.api.BakeRequest
+import com.netflix.spinnaker.rosco.providers.CloudProviderBakeHandler
+import com.netflix.spinnaker.rosco.providers.tencent.config.RoscoTencentConfiguration
+import com.netflix.spinnaker.rosco.providers.util.ImageNameFactory
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Slf4j
+@Component
+public class TencentBakeHandler extends CloudProviderBakeHandler {
```

Comment:
We are slowly deprecating the use of groovy in spinnaker services. We recommend either using `Java` or `Kotlin`.

Reply:
Unless the current structure absolutely forces you to use groovy